marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/mapping-functions/parseDate.sjs

```diff
@@ -0,0 +1,70 @@
+/*
+  Copyright 2012-2019 MarkLogic Corporation
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+*/
+
+function parseDate(picture, value) {
+    let standardFormats = ["MM/DD/YYYY", "DD/MM/YYYY", "MM-DD-YYYY", "MM.DD.YYYY", "DD.MM.YYYY", "YYYYMMDD", "YYYY/MM/DD"];
+    let nonStandardFormats = ["Mon DD, YYYY", "DD Mon YYYY", "DD-Mon-YYYY" ];
+    let response;
+    if(standardFormats.includes(picture.trim())){
+        try{
+            let standardizedDate = xdmp.parseYymmdd(picture.replace("YYYY", "yyyy").replace("DD","dd"), value).toString().split("T")[0];
+            response = xs.date(standardizedDate);
+        }
+        catch (ex) {
+            fn.error(null, "Given value doesn't match with the specified pattern for parsing date string.");
```

Comment:
For each error message, I recommend adding the value that was inputted so the error message is more helpful.